gobblin-runtime/src/main/java/gobblin/runtime/Fork.java

```diff
@@ -134,18 +135,9 @@ public Fork(TaskContext taskContext, Object schema, int branches, int index) thr
       buildWriterIfNotPresent();
     }
 
-    this.recordQueue = BoundedBlockingRecordQueue.newBuilder()
-        .hasCapacity(this.taskState.getPropAsInt(
+    this.recordQueue = new LinkedBlockingQueue<Object>(this.taskState.getPropAsInt(
```

Comment:
Additionally, I don't think we want to used LinkedBlockingQueue as it's performance is worse then ArrayBlockingQueue.